whisper/whisper/timing.py

```diff
@@ -175,40 +122,36 @@ def find_alignment(
     if len(text_tokens) == 0:
         return []
 
-    tokens = torch.tensor(
+    tokens = mx.array(
         [
             *tokenizer.sot_sequence,
             tokenizer.no_timestamps,
             *text_tokens,
             tokenizer.eot,
         ]
-    ).to(model.device)
-
-    # install hooks on the cross attention layers to retrieve the attention weights
-    QKs = [None] * model.dims.n_text_layer
-    hooks = [
-        block.cross_attn.register_forward_hook(
-            lambda _, ins, outs, index=i: QKs.__setitem__(index, outs[-1][0])
-        )
-        for i, block in enumerate(model.decoder.blocks)
-    ]
-
-    with torch.no_grad():
-        logits = model(mel.unsqueeze(0), tokens.unsqueeze(0))[0]
-        sampled_logits = logits[len(tokenizer.sot_sequence) :, : tokenizer.eot]
-        token_probs = sampled_logits.softmax(dim=-1)
-        text_token_probs = token_probs[np.arange(len(text_tokens)), text_tokens]
-        text_token_probs = text_token_probs.tolist()
+    )
 
-    for hook in hooks:
-        hook.remove()
+    logits, _, cross_qk = model(mel[None, :], tokens[None, :])
+    sampled_logits = logits[0][len(tokenizer.sot_sequence) :, : tokenizer.eot]
+    # token_probs = mx.softmax(sampled_logits.astype(mx.float32), axis=-1).astype(sampled_logits.dtype)
+    # todo: np float32 ?
+    token_probs = mx.softmax(sampled_logits.astype(mx.float32), axis=-1)
+    # todo: mlx ?
+    text_token_probs = np.array(token_probs)[np.arange(len(text_tokens)), text_tokens]
```

Comment:
I got an error while doing this in MLX by `token_probs[range(len(text_tokens)), text_tokens]`.

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ValueError: Cannot index mlx array using the given type yet
```

Maybe you have better ideas?

Reply:
I think the `range` is an issue?

`token_probs[list(range(len(text_tokens))), text_tokens]`

Reply:
doesn't work either. I think it's because we can't yet index with list/array? even tried with `token_probs[[0], :]` but got the same error

Reply:
You're absolutely right, I forgot we don't support list indexing 🤦 .

That should work: `token_probs[mx.arange(text_tokens), text_tokens]` but I think you are better off using a `take_along_axis`

Reply:
yeah, thanks!